tardis/energy_input/gamma_ray_grid.py

```diff
@@ -0,0 +1,301 @@
+import numpy as np
+from astropy.coordinates import cartesian_to_spherical
+import re
+from nuclear.io.nndc import get_decay_radiation_database, store_decay_radiation
+import pandas as pd
+
+from tardis.energy_input.util import (
+    solve_quadratic_equation,
+    convert_half_life_to_astropy_units,
+)
+from tardis.montecarlo.montecarlo_numba.numba_config import CLOSE_LINE_THRESHOLD
+from tardis.energy_input.energy_source import load_nndc_decay_data
+
+
+def calculate_distance_radial(gxpacket, r_inner, r_outer):
+    """
+    Calculates 3D distance to shell from gamma ray position
+
+    Parameters
+    ----------
+    gxpacket : GXPacket object
+    r_inner : dtype float
+    r_outer : dtype float
+
+    Returns
+    -------
+    distance : dtype float
+
+    """
+    # determine cartesian location coordinates of gamma-ray object
+    x, y, z = gxpacket.location.get_cartesian_coords
+    # determine cartesian direction coordinates of gamma-ray object
+    x_dir, y_dir, z_dir = gxpacket.direction.get_cartesian_coords
+    # solve the quadratic distance equation for the inner and
+    # outer shell boundaries
+    inner_1, inner_2 = solve_quadratic_equation(
+        x, y, z, x_dir, y_dir, z_dir, r_inner
+    )
+    outer_1, outer_2 = solve_quadratic_equation(
+        x, y, z, x_dir, y_dir, z_dir, r_outer
+    )
```

Comment:
Maybe only calculate distances that are strictly needed instead of all four by default?